### PR TITLE
[2.3.0] Build CUDA 11.8 docker images

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Cherry-pick of [9995599](https://github.com/ray-project/ray/commit/999559998b4f8d6d0b34fe02a4a504823ea1dfc3) without changing default base images. 

We want to cherry-pick this commit so that we provide a base GPU image that can support PyTorch 2.0 (e.g. `rayproject/ray:latest-py38-gpu`)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by zhe-thoughts — APPROVED** [review]
Approving for cherry picking into 2.3.0. But like @krfricke mentioned, let's wait for @amogkam approval as code owner.

**Review by amogkam — APPROVED** [review]

**Review by cadedaniel — APPROVED** [review]

**Pull request merged**
